apis/core/lib/domain/csv-source/CsvSource.ts

```diff
@@ -36,13 +36,14 @@ export default function Mixin<Base extends Constructor<Omit<CsvSource, keyof Api
     @property.literal({ path: schema.error })
     error?: string
 
-    setUploadedFile(key: string, contentUrl: NamedNode): void {
+    setUploadedFile(sourceKind: NamedNode, key: string, contentUrl: NamedNode): void {
       if (this.associatedMedia) {
-        throw new Error('Source file already exists')
+        this.associatedMedia.pointer.deleteOut()
```

Comment:
Is this a good way to delete the previous associated media?

Reply:
This only removes the `schema:associatedMedia` triple. I think you'd also have to go deep into that subgraph to remove the properties of old media object

Reply:
Are you sure? I thought that `deleteOut()` without specifying the property would delete all the properties of the associated media.

Reply:
Of course, I was wrong. Sorry